src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs

```diff
@@ -46,6 +46,48 @@ public virtual bool TryGetFieldType(string name, out FormulaType type)
             return true;
         }
 
+        /// <summary>
+        /// Lookup for logical name and field for input display or logical name.
+        /// If there is a conflict, it prioritizes logical name.
+        /// i.e. field1->Logical=F1 , Display=Display1; field2-> Logical=Display1, Display=Display2
+        /// would return field2 with logical name Display1.
+        /// </summary>
+        /// <param name="displayOrLogicalName">Display or Logical name.</param>
+        /// <param name="logical">Logical name for the input.</param>
+        /// <param name="type">Type for the input Display or Logical name.</param>
+        /// <returns>true or false.</returns>
+        /// <exception cref="ArgumentNullException">Throws, if input displayOrLogicalName is empty.</exception>
+        public bool TryGetFieldType(string displayOrLogicalName, out string logical, out FormulaType type)
+        {
+            if (string.IsNullOrEmpty(displayOrLogicalName))
+            {
+                throw new ArgumentNullException(nameof(displayOrLogicalName));
+            }
+
+            if (_type.DisplayNameProvider.TryGetDisplayName(new DName(displayOrLogicalName), out _))
+            {
+                logical = displayOrLogicalName;
+            }
+            else if (_type.DisplayNameProvider.TryGetLogicalName(new DName(displayOrLogicalName), out var maybeLogical))
+            {
+                logical = maybeLogical;
+            }
+            else
+            {
+                logical = null;
+                type = Blank;
+                return false;
+            }
+
+            if (!TryGetFieldType(logical, out type))
```

Comment:
This block could just be `return TryGetFieldType(logical, out type)`. Not necessary to clear logical before returning, if it's not found in the type, the derived class is violating a contract by having it in the DisplayNameProvider. You could even use the more strict `GetFieldType` if you wanted to verify that.

Reply:
updated with return TryGetFieldType(logical, out type)